ddtrace/opentracer/tracer.py

```diff
@@ -1,13 +1,69 @@
 import opentracing
+import logging
+
+from ddtrace import Tracer as DatadogTracer
+from ddtrace.constants import FILTERS_KEY
+from ddtrace.ext import AppTypes
+from ddtrace.settings import ConfigException
+
+from .settings import ConfigKeys as keys, config_invalid_keys
+
+from .util import merge_dicts
+
+
+log = logging.getLogger(__name__)
+
+DEFAULT_CONFIG = {
+    keys.AGENT_HOSTNAME: 'localhost',
+    keys.AGENT_PORT: 8126,
+    keys.DEBUG: False,
+    keys.ENABLED: True,
+    keys.GLOBAL_TAGS: {},
+    keys.SERVICE_NAME: None,
+    keys.SAMPLER: None,
+    keys.APP_TYPE: AppTypes.worker,
+    keys.CONTEXT_PROVIDER: None,
+    keys.PRIORITY_SAMPLING: None,
+    keys.SETTINGS: {
+        FILTERS_KEY: [],
+    },
+}
 
 
 class Tracer(opentracing.Tracer):
-    """"""
+    """A wrapper providing an OpenTracing API for the Datadog tracer."""
 
-    __slots__ = []
+    __slots__ = ['_enabled', '_debug', '_service_name', '_tracer']
 
-    def __init__(self, scope_manager=None):
-        pass
+    def __init__(self, config={}, service_name=None, scope_manager=None):
+        # Merge the given config with the default into a new dict
+        self._config = merge_dicts(DEFAULT_CONFIG, config)
+
+        # Pull out commonly used properties for performance
+        self._service_name = self._config.get(keys.SERVICE_NAME, None) or service_name
```

Comment:
do we need this kwarg? why we can't put everything under the config object? because it's not clear to the user what happens if they:

```python
t = Tracer(config={'service_name': 'foo'}, service_name='bar')
```

There should be one-- and preferably only one --obvious way to do it.

Reply:
Yes, good point. I agree there should only be one obvious way to do it.

I think I pulled the service name out of the config to be similar with the jaegar client.

There isn't really anything in the opentracing spec with regards to configuration.

Reply:
yes constructors are mostly vendor specific in OpenTracing. If we want to keep the `service_name` out, then let's not use it in the `Config`. I'm fine with this approach since the `service_name` is something that will be changed more often. Other settings may not and defaults are fine.

Reply:
also, let's keep `service_name` as a first kwarg